log4j-redis/src/main/java/org/apache/logging/log4j/redis/appender/RedisAppender.java

```diff
@@ -0,0 +1,202 @@
+
+package org.apache.logging.log4j.redis.appender;
+
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.Node;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginBuilderFactory;
+import org.apache.logging.log4j.core.config.plugins.validation.constraints.Required;
+import org.apache.logging.log4j.core.layout.AbstractStringLayout;
+
+import java.io.Serializable;
+import java.nio.charset.Charset;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sends log events to a Redis Queue. All logs are appended Redis lists via the RPUSH command.
+ */
+@Plugin(name = "Redis", category = Node.CATEGORY, elementType = Appender.ELEMENT_TYPE, printObject = true)
+public final class RedisAppender extends AbstractAppender {
+
+    private RedisAppender(final String name, final Layout<? extends Serializable> layout, final Filter filter,
+                          final boolean ignoreExceptions, final RedisManager manager) {
+        super(name, filter, layout, ignoreExceptions);
+        this.manager = Objects.requireNonNull(manager, "Redis Manager");
+    }
+
+    /**
+     * Builds RedisAppender instances.
+     * @param <B> The type to build
+     */
+    public static class Builder<B extends Builder<B>> extends AbstractAppender.Builder<B>
+            implements org.apache.logging.log4j.core.util.Builder<RedisAppender> {
+
+        @PluginAttribute("keys")
+        private String[] keys;
+
+        @PluginAttribute(value = "host")
+        @Required(message = "No Redis hostname provided")
+        private String host;
+
+        @PluginAttribute(value = "port")
+        private int port;
+
+        @PluginAttribute(value = "ssl")
+        private boolean ssl = false;
+
+        @SuppressWarnings("resource")
+        @Override
+        public RedisAppender build() {
+            return new RedisAppender(getName(), getLayout(), getFilter(), isIgnoreExceptions(), getRedisManager());
+        }
+
+        public Charset getCharset() {
+            if (getLayout() instanceof AbstractStringLayout) {
+                return ((AbstractStringLayout) getLayout()).getCharset();
+            } else {
+                return Charset.defaultCharset();
+            }
+        }
+
+        String[] getKeys() {
+            return keys;
+        }
+
+        String getHost() {
+            return host;
+        }
+
+        boolean getSsl() {
+            return ssl;
+        }
+
+        int getPort() {
+            return port;
+        }
+
+        public B withKeys(final String key) {
+            this.keys = new String[]{key};
+            return asBuilder();
+        }
```

Comment:
I think we should use "set" as the setter prefix instead of "with".

Reply:
@garydgregory I tend to prefer set as well, but the AbstractAppender and other common appenders (i.e. FileAppender, Rolling RandomAccessFileAppender, etc) all seem to use with___. Of course happy to still use set if you prefer.